homeassistant/components/sensor/hddtemp.py

```diff
@@ -0,0 +1,124 @@
+"""
+Support for getting the disk temperature of a host.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.hddtemp/
+"""
+import logging
+from datetime import timedelta
+from telnetlib import Telnet
+
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, CONF_HOST, CONF_PORT, TEMP_CELSIUS, TEMP_FAHRENHEIT,
+    STATE_UNKNOWN)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_DEVICE = 'device'
+ATTR_MODEL = 'model'
+
+DEFAULT_HOST = 'localhost'
+DEFAULT_PORT = 7634
+DEFAULT_NAME = 'HD Temperature'
+DEFAULT_TIMEOUT = 5
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the HDDTemp sensor."""
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+
+    hddtemp = HddTempData(host, port)
+    hddtemp.update()
+
+    if hddtemp.data is None:
+        _LOGGER.error("Unable to fetch the data from %s:%s", host, port)
+        return False
+
+    add_devices([HddTempSensor(name, hddtemp)])
+
+
+class HddTempSensor(Entity):
+    """Representation of a HDDTemp sensor."""
+
+    def __init__(self, name, hddtemp):
+        """Initialize a HDDTemp sensor."""
+        self.hddtemp = hddtemp
+        self._name = name
+        self._state = False
+        self._details = None
+        self.update()
```

Comment:
I think you can remove that after you call it on line 47. That also better if you will port to async later..